core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -198,7 +196,7 @@ private[spark] class IndexShuffleBlockResolver(
     // find out the consolidated file, then the offset within that from our index
     val indexFile = getIndexFile(blockId.shuffleId, blockId.mapId)
 
-    val in = new DataInputStream(Files.newInputStream(indexFile.toPath))
+    val in = new DataInputStream(new FileInputStream(indexFile))
```

Comment:
@jerryshao this is another place. In addition, I'm not sure if there is any compression codec using `skip` or not.

I also noticed `sun.nio.ch.ChannelInputStream` has extra `synchronized`s as `Files.newInputStream` needs to be thread-safe. Not sure if it may cause performance regression or not.

Reply:
I see.